### ISSUE TITLE
Design simpler / more user friendly alert component for mobile

### ISSUE THREAD

**Issue by sshein** [department-of-veterans-affairs/va.gov-team#14485]
## Issue Description
In the Mobile month research findings, we noticed veterans scrolling a lot to get past our alerts, specifically the COVID alerts and the alerts on content pages telling them they can login to fill out a form. 

Sometimes the alerts confused users because they switched content pages but nothing seemed to have changed, since the alerts were long and stayed at the top.

Let's design something that's lighter weight (can we collapse some content in the notification letting folks know they can login?) and potentially eliminate some of the padding.

---
## Tasks
- [ ] _What work is necessary for this story to be completed?_

## Acceptance Criteria
- [ ] _What will be created or happen as a result of this story?_

**Comment by andaleliz** [comment]
@peggygannon this is the alert ticket we discussed today. Let the condensing begin!

**Comment by andaleliz** [comment]
@sshein Should this be assigned to Crystabel as the design system designer, or is this up to one of the members of our VSA UX team to pick up?

**Comment by sshein** [comment]
good question. I think this should be sent to the design system team and triaged there somehow. Not sure how they want to do that. Maybe @caw310 and @CrystabelReiter can decide (for context this was one of the tickets that came out of the mobile month study)

**Comment by sshein** [comment]
That said, I know @KevinMHoffmanUSDS recently said the design system team does not create new things - only documents them like a library? So maybe they wouldn't do this? 🤔

**Comment by andaleliz** [comment]
I recall that statement as well...but this is an improvement to an existing team. Still, librarians don't edit books so sounds like it would go to VSA.  I'll wait to hear back from them.

**Comment by CrystabelReiter** [comment]
@sshein yes you're right, design system team does not create new solutions. 

If a VSA team member wants to create a mobile friendly alert and wants to propose to update the pattern, then a proposal would be sent to the design system council to review and it will be prioritized for the design system team to work on

**Comment by sshein** [comment]
this is actually an interesting use case - basically my hunch is that no VSA team would prioritize this, because it's not required as part of a product. And VFS teams are very delivery focused. But it's something we identified as a component improvement from mobile month. I wonder how realistic it is to rely on VSF teams to always be the ones improving components.

**Comment by andaleliz** [comment]
I can see that @sshein.  In the past, when I've included improvements to global components in designs for my team, I have been told we can't prioritize those improvements since they'd be a global change, and we should just move forward with what we have.  I'd like to work on this but it's not part of any current initiatives we have.

**Comment by andaleliz** [comment]
@jonathanjnelson has an alert in a flow he is working on and is willing to propose some mobile improvements as an extension of that work!

**Comment by CrystabelReiter** [comment]
@sshein @andaleliz These are both good points. I think relying solely on VFS teams for improving components would be a good topic to bring to design system council discussion on Friday. 

Looking forward to seeing Jonathan's mobile improvements!

**Comment by andaleliz** [comment]
assigning myself so this doesn't get lost in the shuffle; hoping another designer can tackle

**Comment by andaleliz** [comment]
@rileyorr thanks for volunteering to take this on! I dropped the lightweight alert sketch file that Shawna started in our VSA design files folder - [hopefully you can access it through this link](https://drive.google.com/drive/folders/18Hk_kqwQpJDlef6JR6CfV_lJpHZ3k65t?usp=sharing).  If not, lmk and I'll attach to the ticket (I try to avoid that when possible to minimize things getting out of sync)

I also found t[his alert exploration @bridgethapner did for mobile month](https://dsva.slack.com/archives/C0NGDDXME/p1597686819278400). I'm checking with her to see if any of it was ever tested, and to see how the mobile team is handling alerts as it may be of interest - will post here if there is more to share!

**Comment by andaleliz** [comment]
@rileyorr I heard from Bridget; they haven't done any explicit testing of alerts on mobile, but users have encountered their error alert pattern in usability studies and they haven't seen any issues.  Attached are some mock-ups of how they're handling alerts in the app: mostly the same way the web does, just without the icon and with less padding to give the text more space on mobile.
<img width="586" alt="Screen Shot 2021-03-17 at 4 35 18 PM" src="https://user-images.githubusercontent.com/55992116/111784214-82248900-8891-11eb-8a19-81bfe2023f4c.png">

**Comment by lildziner** [comment]
@rileyorr @sshein wanted to give an update regarding our team's expandable version of a facility operating status alert ([Issue 24195](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24195)). Our goal is to include this as part of the launch of Vet Center MVP product page sites. We had to take an MVP approach to it's design and not allow rich text. Tappable phone will be handled by FE and not built into the CMS. We plan to revisit this in the future after we see how they perform.

**Comment by andaleliz** [comment]
@lildziner This alert is super cool! I have a few questions:
1.  Once you see how they perform, are you planning to propose for it to be adopted into the designs system, assuming it performs well?
2. Can you elaborate on what you mean by "not allow rich text"? Is that just because it's a CMS component? It looks like there's HTML in it, based on your mock-ups.
3. Any ETA on when in the future you'd revisit?

**Comment by lildziner** [comment]
@andaleliz thanks!
> 1. Once you see how they perform, are you planning to propose for it to be adopted into the designs system, assuming it performs well?
We want to learn from this version and see how it might need to evolve for the design system but yes, the plan is for it to go in the experimental design system and go through the vetting process. 
> 2. Can you elaborate on what you mean by "not allow rich text"? Is that just because it's a CMS component? It looks like there's HTML in it, based on your mock-ups.
Currently, this version would not allow bullets, bold, or adding in hyperlinks via the CMS. For now just plain text. Until we can come up with very strict usage guidelines we want to avoid the 'wall of COVID' problem that we have on VAMC global alerts, which causes wayfinding issues for users. The hyperlinked phone number you see in the mockup is actually achieved by the front end via a plug-in that's currently already in use.
> 3. Any ETA on when in the future you'd revisit?
Good question. The experimental design system process might drive this timeline to some degree but also we will want to see what usage stats we can get from the VC pilot launches. e.g. do people actually expand them?

**Comment by rileyorr** [comment]
@lildziner I wanted to give an update since the debt resolution team just tested this component during the medical copay usability study! Here's a snippet of our prototype that included the alert:

![mobile alert](https://user-images.githubusercontent.com/46329147/127016446-e9419ba3-d5de-4823-b73f-a4af816ffeac.png)

**Our findings:**
- The alert was often the first thing Veterans noticed when arriving on the page
- Some clicked on the alert to read more about the referred charge while others moved on to looking at either their balance details or resolution options 
- Veterans generally understood that they could interact with the alert. However, one asked themselves, "Can I click on this?" before clicking on it but that may have been because they were interacting with a prototype where not every element was interactive rather than because they were unsure that the alert was clickable in general

---

This component is also currently live in [the debt portal](https://staging.va.gov/manage-va-debt/your-debt/). 

![75b alert in staging](https://user-images.githubusercontent.com/46329147/127017746-1efea931-0483-45d3-84a3-b54c44a1f103.png)

@LinnJS referenced [the Denver Vet Center alert](https://www.va.gov/colorado-springs-vet-center/locations/) in staging, but found that the code wasn't very reusable. I think when he implemented the 75b alert in the debt portal he made adjustments to make the code reusable. If so, it would be worth syncing up so his code can be incorporated into your writeup for the experimental design system!

**Comment by lildziner** [comment]
Thanks for sharing this update @rileyorr!  And to your point, I think your version of the expandable alert with the adjusted code might be a better example for the experimental design system than the one I tried to pull. @sshein I think this is what you were referring to?

**Comment by sshein** [comment]
yep that's what I was referring to :)
